src/messaging/ExchangeMgr.cpp

```diff
@@ -112,7 +113,10 @@ CHIP_ERROR ExchangeManager::Shutdown()
 
 ExchangeContext * ExchangeManager::NewContext(SecureSessionHandle session, ExchangeDelegate * delegate)
 {
-    return AllocContext(mNextExchangeId++, session, true, delegate);
+    uint16_t nextId = mNextExchangeId++;
+    if (nextId == kReservedExchangeId)
```

Comment:
The kReservedExchangeId is defined to 0xFFFF, it can not be incremented in this case.

Reply:
Hi Mingjie,
Could you also address this comment? I don't we can do another incremental if nextId == 0xFFFF

Reply:
@kghost, is this intentional (i.e. change the nextId to 0 if it reaches 0xFFFF)?

Reply:
Yes,it is. It is the same logic as before but with the 0xFFFF value being reserved for legacy usage. 

Additional check may needed to avoid exchange Id conflict then it loops around, but it should be another PR instead.